common/client-core/src/client/base_client/mod.rs

```diff
@@ -356,6 +359,11 @@ where
         self
     }
 
+    pub fn with_connect_timeout(mut self, timeout: Duration) -> Self {
```

Comment:
I think it would it be easier for you if rather than setting connection timeout as a separate thing of the client, you put it as part of the base client config. `gateway_connection` section inside the `DebugConfig` sounds like the perfect place for it, wouldn't you say?